biogeochem/EDLoggingMortalityMod.F90

```diff
@@ -168,41 +185,92 @@ end subroutine IsItLoggingTime
+         ! Pass logging rates to cohort level 
+         
+         if (hlm_use_lu_harvest == ifalse) then
+            ! 0=use fates logging parameters directly when logging_time == .true.
+            ! this means harvest the whole cohort area
+            harvest_rate = 1._r8
+            
+         else if (hlm_use_lu_harvest == itrue .and. hlm_harvest_units == hlm_harvest_area_fraction) then
+            ! We are harvesting based on areal fraction, not carbon/biomass terms. 
+            ! 1=use area fraction from hlm
+            ! combine forest and non-forest fracs and then apply:
+            ! primary and secondary area fractions to the logging rates, which are fates parameters
+            
+            ! Definitions of the underlying harvest land category variables
+            ! these are hardcoded to match the LUH input data via landuse.timseries file (see dynHarvestMod)
+            ! these are fractions of vegetated area harvested, split into five land category variables
+            ! HARVEST_VH1 = harvest from primary forest
+            ! HARVEST_VH2 = harvest from primary non-forest
+            ! HARVEST_SH1 = harvest from secondary mature forest
+            ! HARVEST_SH2 = harvest from secondary young forest
+            ! HARVEST_SH3 = harvest from secondary non-forest (assume this is young for biomass)
+
+            ! Get the area-based harvest rates based on info passed to FATES from the boundary condition
+            call get_harvest_rate_area (patch_anthro_disturbance_label, hlm_harvest_catnames, &
+                 hlm_harvest_rates, frac_site_primary, secondary_age, harvest_rate)
+
+         else if (hlm_use_lu_harvest == itrue .and. hlm_harvest_units == hlm_harvest_carbon) then
+            ! 2=use carbon from hlm
+            ! not implemented yet
+            write(fates_log(),*) 'HLM harvest carbon data not implemented yet. Exiting.'
+            call endrun(msg=errMsg(sourcefile, __LINE__))
+         endif
+
+         ! transfer of area to secondary land is based on overall area affected, not just logged crown area
+         ! l_degrad accounts for the affected area between logged crowns
          if(EDPftvarcon_inst%woody(pft_i) == 1)then ! only set logging rates for trees
-
-            ! Pass logging rates to cohort level 
-
-            if (dbh >= logging_dbhmin ) then
-               lmort_direct = logging_direct_frac * adjustment
-               l_degrad = 0._r8
+            
+            if (dbh >= logging_dbhmin .and. .not. &
+                 ((logging_dbhmax < fates_check_param_set) .and. (dbh < logging_dbhmax )) ) then
+               lmort_direct = harvest_rate * logging_direct_frac
+               l_degrad = harvest_rate * (1._r8 - logging_direct_frac) ! fraction passed to 'degraded' forest. 
             else
-               lmort_direct = 0.0_r8 
-               l_degrad = logging_direct_frac * adjustment
+               lmort_direct = 0.0_r8
+               l_degrad = harvest_rate
```

Comment:
Does this mean that if trees are not large enough to log, the cohort is still degraded/moved to secondary patch?

Reply:
Oh, never mind I see, background degradation rate regardless of logging.